src/OrchardCore.Modules/OrchardCore.Secrets.KeyVault/Services/KeyVaultClientService.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Options;
+using OrchardCore.Secrets.KeyVault.Models;
+
+namespace OrchardCore.Secrets.KeyVault.Services;
+
+public class KeyVaultClientService
+{
+    private readonly SecretClient _client;
+    private readonly string _prefix;
+
+    public KeyVaultClientService(IOptions<SecretsKeyVaultOptions> options)
+    {
+        var keyVaultEndpointUri = new Uri($"https://{options.Value.KeyVaultName}.vault.azure.net");
+
+        _client = new SecretClient(
+            keyVaultEndpointUri,
+            new DefaultAzureCredential(new DefaultAzureCredentialOptions { ExcludeVisualStudioCodeCredential = true }));
+
+        _prefix = options.Value.Prefix;
+    }
+
+    public async Task<string> GetSecretAsync(string name)
+    {
+        if (!string.IsNullOrEmpty(_prefix))
+        {
+            name = $"{_prefix}{name}";
+        }
+
+        var secret = await _client.GetSecretAsync(name);
```

Comment:
The same as from any secrets provider but I don't remember what happens in that case.

Yes I thought about fallback to a plain text if it exists but as I remember I just let it failing, what you think?

Reply:
`SecretService` returns an empty secret if the return value here is `null`, so I'd follow that pattern and return `null` here instead of failing with an exception.

Reply:
Okay